pkg/cmd/cli/cmd/loginoptions.go

```diff
@@ -0,0 +1,325 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+	"github.com/openshift/origin/pkg/client"
+
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	kclientcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+	"github.com/openshift/origin/pkg/cmd/cli/config"
+	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	"github.com/openshift/origin/pkg/user/api"
+)
+
+const defaultClusterURL = "https://localhost:8443"
+
+// Helper for the login and setup process, gathers all information required for a
+// successful login and eventual update of config files.
+// Depending on the Reader present it can be interactive, asking for terminal input in
+// case of any missing information.
+type LoginOptions struct {
+	// flags and printing helpers
+	Username string
+	Password string
+	Project  string
+
+	// infra
+	ClientConfig clientcmd.ClientConfig
```

Comment:
Based on the usage I see, I think this will be easier to read pulling the you actually care about, `currentlyAvailableMergedConfig`, `currentKClientConfig`, and `defaultNamespace` out.  It will help focus this logic on a smaller set of possible input and prevent the temptation side-effecty code for mutating what the clientcmd.ClientConfig interface returns back.